Source/WebCore/platform/graphics/FontSizeAdjust.h

```diff
@@ -69,12 +69,23 @@ struct FontSizeAdjust {
             : std::nullopt;
     }
 
+    bool resolveFromFontIfNeeded(float computedSize, const FontMetrics& fontMetrics)
+    {
+        if (!shouldResolveFromFont())
+            return false;
+        value = computeValue(computedSize, fontMetrics);
+        hasFromFontResolved = true;
+        return true;
+    }
+
     bool isNone() const { return !value && type != ValueType::FromFont; }
     bool isFromFont() const { return type == ValueType::FromFont; }
+    bool shouldResolveFromFont() const { return isFromFont() && !hasFromFontResolved; }
 
     Metric metric { Metric::ExHeight };
     ValueType type { ValueType::Number };
     Markable<float, WTF::FloatMarkableTraits> value { };
+    bool hasFromFontResolved { false };
```

Comment:
I think this might be better with the name `hasResolvedFromFont`.

Reply:
We do not need this `bool` to indicate the font has been resolved -- `Markable` is a flavor of `std::optional`, so we just need to check it as a `bool`, or `value_or`, or whatever.